analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -61,9 +61,30 @@ protected override ProgramState PreProcess(ProgramState state, IBinaryOperationW
     {
         BinaryOperatorKind.Add => NumberConstraint.From(left.Min + right.Min, left.Max + right.Max),
         BinaryOperatorKind.Subtract => NumberConstraint.From(left.Min - right.Max, left.Max - right.Min),
+        BinaryOperatorKind.Multiply => CalculateMultiply(left, right),
         _ => null
     };
 
+    private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberConstraint right)
+    {
+        var products = new[] { left.Min * right.Min, left.Min * right.Max, left.Max * right.Min, left.Max * right.Max };
+        var min = products.Min();
+        var max = products.Max();
+
+        if ((left.Min is null && right.CanBePositive) || (right.Min is null && left.CanBePositive)
+            || (left.Max is null && right.CanBeNegative) || (right.Max is null && left.CanBeNegative))
+        {
+            min = null;
+        }
+        if ((left.Min is null && right.CanBeNegative) || (right.Min is null && left.CanBeNegative)
+            || (left.Max is null && right.CanBePositive) || (right.Max is null && left.CanBePositive))
+        {
+            max = null;
+        }
+
```

Comment:
Not needed, there's already semicolon, EOL, closing bracket and EOL as a command separator :)
```suggestion
```